### DISCUSSION TITLE
How to authenticate an automated service to Supabase?

### DISCUSSION THREAD

**Discussion by carlesandres** [supabase/supabase#22704] [Questions]
I want to do the following:

1. Run Vercel CRON job which
2. Calls a Next.js API endpoint which
3. Makes a call to a table in Supabase (protected by RLS)

The API endpoint is public but already protected by a password that's only set in Vercel's cron jobs.  I would want to sign into Supabase from my Vercel function with the special user that has access to my table, since using the service role key seems too powerful and risky. But I am not sure the best way to do this.

What's best practice? Do people just create a user with and sign in with `supabase.auth.signInWithPassword`? Are there any other obvious mechanism I am missing for the authentication? The docs for supabase-j2 v2 don't show any `signInWithJwt` method afaik.

**Comment by GaryAustin1** [comment]
You can mint jwt either as authenticated user (role=authenticated and a sub claim) or with just a role and create a postgres role that will run the query.
You set the jwt in the authorization header with createClient and the jwt will be passed into the REST API.
https://supabase.com/docs/guides/auth/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase

**Reply by carlesandres** [reply]
Thanks @GaryAustin1 ! I think I understand 90% of this, but it would be great to see a full example of a jwt being mint with just a role (and the simplest way to create a role in Supbase). Are there any examples of videos covering that?

**Reply by GaryAustin1** [reply]
Sorry, I've not seen any.
You can use jwt.io to mint a role with the SB secret.
This shows creating a role:
https://supabase.com/docs/guides/database/postgres/roles

This discussion talks about what has to occur to use a new role from the REST API (assigning to authenticator role)
https://github.com/orgs/supabase/discussions/11948
